plugins/cpp_metrics/parser/src/cppmetricsparser.cpp

```diff
@@ -0,0 +1,78 @@
+#include <cppmetricsparser/cppmetricsparser.h>
+
+#include <model/cppastnodemetrics.h>
+#include <model/cppastnodemetrics-odb.hxx>
+#include <model/cppdirectorymetrics.h>
+#include <model/cppdirectorymetrics-odb.hxx>
+
+#include <boost/filesystem.hpp>
+
+#include <util/logutil.h>
+
+#include <memory>
+
+namespace cc
+{
+namespace parser
+{
+
+CppMetricsParser::CppMetricsParser(ParserContext& ctx_): AbstractParser(ctx_)
+{
+}
+
+bool CppMetricsParser::accept(const std::string& path_)
+{
+  std::string ext = boost::filesystem::extension(path_);
+  return ext == ".dummy";
+}
+
+bool CppMetricsParser::parse()
+{        
+  for(std::string path : _ctx.options["input"].as<std::vector<std::string>>())
+  {
+    if(accept(path))
+    {
+      LOG(info) << "CppMetricsParser parse path: " << path;
+    }
+  }
+  return true;
+}
+
+CppMetricsParser::~CppMetricsParser()
+{
+}
+
+/* These two methods are used by the plugin manager to allow dynamic loading
+   of CodeCompass Parser plugins. Clang (>= version 6.0) gives a warning that
+   these C-linkage specified methods return types that are not proper from a
+   C code.
+
+   These codes are NOT to be called from any C code. The C linkage is used to
+   turn off the name mangling so that the dynamic loader can easily find the
+   symbol table needed to set the plugin up.
+*/
+// When writing a plugin, please do NOT copy this notice to your code.
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wreturn-type-c-linkage"
+extern "C"
+{
+  boost::program_options::options_description getOptions()
+  {
+    boost::program_options::options_description description("Cpp Metrics Plugin");
+
+    description.add_options()
+        ("dummy-arg", po::value<std::string>()->default_value("Dummy arg"),
+          "This argument will be used by the dummy parser.");
+
+    return description;
+  }
```

Comment:
If there are no CLI arguments, this could be simplified to:
```cpp
boost::program_options::options_description getOptions()
{
  boost::program_options::options_description description("Cpp Metrics Plugin");
  return description;
}
```

Reply:
Fixed in a904b68c9f6b79f09bb04a1a1461cec0d803af9e.

Reply:
It was fixed in 8fca9f6754a68d4ba53416c6637647fcdeb309f4, but ok.